legato-api/src/main/java/org/opendaylight/legato/LegatoServiceController.java

```diff
@@ -0,0 +1,384 @@
+import org.opendaylight.yang.gen.v1.urn.onf.params.xml.ns.yang.tapi.connectivity.rev171113.DeleteConnectivityServiceInput;
+import org.opendaylight.yang.gen.v1.urn.onf.params.xml.ns.yang.tapi.connectivity.rev171113.DeleteConnectivityServiceInputBuilder;
+import org.opendaylight.yang.gen.v1.urn.onf.params.xml.ns.yang.tapi.connectivity.rev171113.UpdateConnectivityServiceInput;
+import org.opendaylight.yang.gen.v1.urn.onf.params.xml.ns.yang.tapi.connectivity.rev171113.UpdateConnectivityServiceOutput;
+import org.opendaylight.yang.gen.v1.urn.onf.params.xml.ns.yang.tapi.connectivity.rev171113.TapiConnectivityService;
+import org.opendaylight.yang.gen.v1.urn.mef.yang.mef.types.rev171215.EvcIdType;
+
+import org.opendaylight.yangtools.concepts.ListenerRegistration;
+import org.opendaylight.yangtools.yang.binding.InstanceIdentifier;
+import org.opendaylight.yangtools.yang.common.RpcResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+
+public class LegatoServiceController extends UnimgrDataTreeChangeListener<Evc> {
+
+	public LegatoServiceController(DataBroker dataBroker) {
+		super(dataBroker);
+		registerListener();
+	}
+
+	private static final Logger LOG = LoggerFactory
+			.getLogger(LegatoServiceController.class);
+
+	private static final InstanceIdentifier<Evc> EVC_IID = InstanceIdentifier
+			.builder(MefServices.class).child(CarrierEthernet.class)
+			.child(SubscriberServices.class).child(Evc.class).build();
+
+	private static final InstanceIdentifier<SubscriberServices> EVC_IID_OPERATIONAL = InstanceIdentifier
+			.builder(MefServices.class).child(CarrierEthernet.class)
+			.child(SubscriberServices.class).build();
+
+	private ListenerRegistration<LegatoServiceController> dataTreeChangeListenerRegistration;
+
+	private static final Map<String, String> EVC_UUIDMap = new HashMap<String, String>();
+
+	private TapiConnectivityService prestoConnectivityService;
+
+	public void setPrestoConnectivityService(
+			TapiConnectivityService prestoConnectivityService) {
+		this.prestoConnectivityService = prestoConnectivityService;
+	}
+
+	public void registerListener() {
+		LOG.info("Initializing LegatoServiceController:int() ");
+
+		assert prestoConnectivityService != null;
+
+		dataTreeChangeListenerRegistration = dataBroker
+				.registerDataTreeChangeListener(new DataTreeIdentifier<Evc>(
+						LogicalDatastoreType.CONFIGURATION, EVC_IID), this);
+	}
+
+	public void close() throws Exception {
+		if (dataTreeChangeListenerRegistration != null) {
+			dataTreeChangeListenerRegistration.close();
+		}
+	}
+
+	@Override
+	public void add(DataTreeModification<Evc> newDataObject) {
+		if (newDataObject.getRootNode() != null
+				&& newDataObject.getRootPath() != null) {
+			LOG.info("  Node Added  "
+					+ newDataObject.getRootNode().getIdentifier());
+
+			Optional<Evc> OptionalEvc = LegatoUtils.readEvc(dataBroker,
```

Comment:
Actually you do not need to do an extra lookup to MD-SAL -  tree modification has all data you need.
``Evc evc = newDataObject.getRootNode().getDataAfter();`` 
Please consider using this apporach elsewhere. The contract is:
* ``getDataAfter`` has value for create and update
* ``getDataBefore`` has value for delete and update
in case of update getDataBefore stores old and getDataAfter a new version of the object